### PR TITLE
Add cfgfile version 0.2.9.1.

### PR DESCRIPTION
**Pull request by igormironchik**
Specify library name and version:  **cfgfile/0.2.9.1**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

I'm the author of this library.

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by igormironchik** [comment]
Guys, I need your help. I don't know why built `cfgfile.generator` wasn't found by build of test.

Installation of cfgfile reported:

```
cfgfile/0.2.9.1: Appending PATH env var with : /Users/jenkins/w/BuildSingleReference/.conan/data/cfgfile/0.2.9.1/_/_/package/b308dde9a45ef5a1363975ca1ddfc2b271fd37d9/bin
```

Where `cfgfile.generator` places.

**Comment by SpaceIm** [comment]
Why so many folders in this recipe? Can't you merge all of them?

**Comment by igormironchik** [comment]
I thought about it. Could you clarify my doubts. Version 0.2.8.2.1 doesn't have any requirements. Version 0.2.9.0 requires args-parser 6.0.1.0. Version 0.2.9.1 requires args-parser 6.2.0.1. Versions 0.2.9.0 and 0.2.9.1 use

```py
self._cmake.definitions["USE_INTERNAL_ARGS_PARSER"] = False
```

I'm not an expert of Conan. Is it possible to handle all these differences?

**Comment by SpaceIm** [comment]
Sure, you can add branching depending on library version.

**Comment by igormironchik** [comment]
Didn't understand. Could you show me example? What do you mean by adding branching?

**Comment by SpaceIm** [comment]
```python
if tools.Version(self.version) >= "6.2.0.1":
    # whatever logic
```

**Comment by igormironchik** [comment]
Ok, I see. Thank you. Just misunderstood because of my English.

File: recipes/cfgfile/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -21,7 +21,7 @@ add_executable(${PROJECT_NAME}
     "${CMAKE_CURRENT_BINARY_DIR}/cfg.hpp"
 )
 
-target_link_libraries(${PROJECT_NAME} cfgfile)
+target_link_libraries(${PROJECT_NAME} cfgfile::cfgfile)
```
Comment (madebr) [inline recipes/cfgfile/all/test_package/CMakeLists.txt:24]: The previous 0.2.9.0 recipe contained logic to recreate a `cfgfile` target (`_create_cmake_module_alias_targets`).
Is this removal intended?

Comment (igormironchik) [reply recipes/cfgfile/all/test_package/CMakeLists.txt:24]: Yes. I don't see a need for it. But very old versions of `cfgfile` (even before Conan packaging) used its name without namespace. What do you think?

Comment (madebr) [reply recipes/cfgfile/all/test_package/CMakeLists.txt:24]: You're the author of this library, so you're the authority.
It makes sense to remove it to get rid of old cruft.
Otoh, cmake has a property for that: [`DEPRECATION`](https://cmake.org/cmake/help/latest/prop_tgt/DEPRECATION.html).

Comment (igormironchik) [reply recipes/cfgfile/all/test_package/CMakeLists.txt:24]: Let it be so, the user of this library should use `cfgfile::cfgfile` target. I do so in all my libraries. I believe that it's minor change for that users who already use `cfgfile` with target without namespace. `DEPRECATION` was introduced in CMake 3.17, but I prefer to support 3.16, as it's default CMake in Ubuntu 20.04.

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM
